comp/core/config/config.go

```diff
@@ -24,16 +24,40 @@ type cfg struct {
 	warnings *config.Warnings
 }
 
+type configDependencies interface {
+	getParams() *Params
+}
+
 type dependencies struct {
 	fx.In
 
 	Params Params
 }
 
-type mockDependencies struct {
-	fx.In
+func (d dependencies) getParams() *Params {
+	return &d.Params
+}
 
-	Params MockParams
+// TODO: serverless must be eventually migrated to fx, this workaround
+//
+//	will then become obsolete - ts should not be created
+//	directly in this fashion.
+func NewServerlessConfig(path string) (Component, error) {
+	options := []func(*Params){WithConfigName("serverless"), WithConfigLoadSecrets(true)}
```

Comment:
What's that side effect of `WithConfigLoadSecrets(true)` here?

Reply:
It's an option so that if `ENC[]` secrets are defined in the agent config, these are resolved when bootstrapping the config if secrets are enabled in the config.